### PR TITLE
Fix parameter value generator and restructure tests

### LINKED ISSUE #29: Update all the links once the repository is moved
README.md
CONTRIBUTING.md
run-simulations.ipynb

### LINKED ISSUE #32: Check parameter_value_generator for functions
At the moment `parameter_value_generator` just returns `param_value`, but I think if the parameter is a function I think you want it to return that function scaled by `param_value`, i.e. `FunctionLike(params[parameter], param_value)`. It looks like for functions it just returns a random number between 0.5 and 2 but maybe I am wrong?

### LINKED ISSUE #33: Clean up the check conditions in `comparison_generator` by putting some acceptable bounds
As pointed out in #30, the code for checking the values of an individual parameter is quite messy and inefficient, it would be good to update that by adding acceptable bounds to a parameter in some way.

### PR DESCRIPTION
**Pull request by Saransh-cpp**
Closes #32 
Closes #33 
Closes #29

## Changes/Additions
1. Fixed parameter_value_generator for functions
2. Added bounds to parameter values
3. Fixed all the links
4. Refactored comparison_generator
5. Restructure the tests into 2 directories, `with_keys` and `without_keys` to run on PRs from this repository and only `without_keys` to run on PRs from forks.

### CODE REVIEW

File: bot/utils/parameter_value_generator.py [hunk 1]
```diff
@@ -12,29 +12,47 @@ def __init__(self, fun, parameter):
...
-        params[parameter] = pybamm.InputParameter(parameter)
-
-    param_value = desired_decimal_point_generator(
-        (base_value[parameter])*0.5, (base_value[parameter])*2, 2
-    )
-
-    return param_value
+        base_value = params[parameter]
+        new_parameter_value = desired_decimal_point_generator(
+            lower_bound if lower_bound is not None else base_value*0.5,
```
Comment (brosaplanella) [inline bot/utils/parameter_value_generator.py:52]: If `base_value` < 1 then `lower_bound` is larger than `upper_bound`. Conceptually should be an issue, but might cause issues with the code.

**Review by brosaplanella — APPROVED** [review]
Looks good, just see my comment on the bounds. Thanks Saransh!

File: bot/utils/parameter_value_generator.py (hunk 1 above)
Comment (Saransh-cpp) [reply bot/utils/parameter_value_generator.py:52]: Taking 0.5 as the `base_value`, the `lower_bound` comes out to be `0.25` and the `upper_bound` comes out to be `1`. Am I doing something wrong or missing something here?

Comment (Saransh-cpp) [reply bot/utils/parameter_value_generator.py:52]: Oh wait, I think negatives will give a problem

Comment (Saransh-cpp) [reply bot/utils/parameter_value_generator.py:52]: Just rechecked, doing this -
```py
round(random.uniform(-0.5, -2), 2)
```
and this - 
```py
round(random.uniform(2, 1), 2)
```
works same as `(-2, -0.5)`, `(1, 2)`

**Comment by Saransh-cpp** [comment]
I see the problem now, the tests are failing because PRs made from a fork are not able to access repository secrets. I'll see if I can fix it and then will re-run the tests.

**Comment by Saransh-cpp** [comment]
@tinosulzer @brosaplanella @rtimms the last 10 commits are about the restructuring of tests (running all the tests on PRs from this repository and skipping a few on PRs from a fork), could you please review them too (Changing the workflow and dividing the tests into 2 sub-directories, `with_keys` and `without_keys`)

**Review by brosaplanella — APPROVED** [review]
Looks good, thanks Saransh!

**Pull request merged**